aries_vcx/src/protocols/issuance/issuer/states/offer_set.rs

```diff
@@ -28,14 +34,45 @@ impl OfferSetState {
     }
 }
 
-impl From<OfferSetState> for OfferSentState {
+impl From<OfferSetState> for FinishedState {
     fn from(state: OfferSetState) -> Self {
-        trace!("SM is now in OfferSent state");
-        OfferSentState {
+        trace!("SM is now in Finished state");
+        FinishedState {
+            cred_id: None,
+            revocation_info_v1: Some(RevocationInfoV1 {
+                cred_rev_id: None,
+                rev_reg_id: state.rev_reg_id,
+                tails_file: state.tails_file,
+            }),
+            status: Status::Undefined,
+        }
+    }
+}
+
+impl From<(OfferSetState, RequestCredential)> for RequestReceivedState {
+    fn from((state, request): (OfferSetState, RequestCredential)) -> Self {
+        trace!("SM is now in Request Received state");
+        RequestReceivedState {
             offer: state.offer,
             cred_data: state.credential_json,
             rev_reg_id: state.rev_reg_id,
             tails_file: state.tails_file,
+            request,
+        }
+    }
+}
+
+impl From<(OfferSetState, ProblemReport)> for FinishedState {
+    fn from((state, err): (OfferSetState, ProblemReport)) -> Self {
+        trace!("SM is now in Finished state");
+        FinishedState {
+            cred_id: None,
+            revocation_info_v1: Some(RevocationInfoV1 {
+                cred_rev_id: None,
+                rev_reg_id: state.rev_reg_id,
+                tails_file: state.tails_file,
+            }),
+            status: Status::Failed(err),
```

Comment:
Any chance of getting rid of conversions like these, or are they outside the scope of this PR? I think a constructor like `FinishedState::new()` is much clearer than something like `(offer_state, problem_report).into()`

Reply:
oh yeah. I hate those from heart :-D Yeap I'll do it here

Reply:
Removed